app/src/main/java/com/iitdh/sonusourav/instigo/Account/LoginActivity.java

```diff
@@ -380,13 +384,28 @@ public void onComplete(@NonNull Task<Void> task) {
                                         }
                                     });
 
-
                                 }else {
+                                    userRef = rootRef.child("Users").child(encodeUserEmail(user.getEmail())).child("name");
+                                    userRef.addListenerForSingleValueEvent(new ValueEventListener() {
+                                        @Override
+                                        public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                                            String name = (String) dataSnapshot.getValue();
+                                            UserProfileChangeRequest request = new UserProfileChangeRequest.Builder()
+                                                    .setDisplayName(name)
+                                                    .build();
+                                            user.updateProfile(request);
+                                        }
+
+                                        @Override
+                                        public void onCancelled(@NonNull DatabaseError databaseError) {
+
+                                        }
+                                    });
+
                                     Toast.makeText(LoginActivity.this, " Signed In as " + user.getDisplayName(), Toast.LENGTH_SHORT).show();
```

Comment:
Still when the user login in for the first time using Google sign-in after he has logged in earlier with Email & password, Toast uses Google username.

Also, only after in the nav bar Google username is used for the first time when signed in using Google.

Reply:
Done